sycl/test-e2e/InOrderEventsExt/set_external_event.cpp

```diff
@@ -55,3 +52,52 @@ int main() {
   free(HostData);
   return Failures;
 }
+
+int check_wait(sycl::queue &Q1, sycl::queue &Q2) {
+  std::cout << "Checking ext_oneapi_set_external_event with wait on queue."
+            << std::endl;
+
+  sycl::buffer<int> DevDataBuf{sycl::range{N}};
+  sycl::accessor DevData{DevDataBuf};
+  int *HostData = (int *)malloc(N * sizeof(int));
```

Comment:
Shouldn't we check for errors here? And also in `check_work`. Alternatively, we may use `new []` operator.

Reply:
Good shout! It has been amended.